### ISSUE TITLE
Samsung Galaxy Grand I9082 not working

### ISSUE THREAD

**Issue by prateekagarwal60** [openstf/stf#253]
Hi,

I connected a Samsung Galaxy Grand I9082 to my setup. And while I'm able to control the device remotely, the screen shown on my laptop is blank and just shows black. I'm getting the following logs:

INF/device:plugins:screen:stream 13306 [4100c76794769000] Requesting frame producer to start
INF/device:plugins:screen:stream 13306 [4100c76794769000] Launching screen service
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "PID: 6294"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: Using projection 480x800@317x528/0"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (external/MY_minicap/src/minicap_17.cpp:239) Creating SurfaceComposerClient"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (external/MY_minicap/src/minicap_17.cpp:242) Performing SurfaceComposerClient init check"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (external/MY_minicap/src/minicap_17.cpp:249) Creating virtual display"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (external/MY_minicap/src/minicap_17.cpp:255) Creating CPU consumer"
INF/device:plugins:screen:stream 13306 [4100c76794769000] Connecting to minicap service
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (external/MY_minicap/src/minicap_17.cpp:259) Creating buffer queue"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (external/MY_minicap/src/minicap_17.cpp:265) Creating frame waiter"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (external/MY_minicap/src/minicap_17.cpp:269) Publishing virtual display"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (jni/minicap/JpgEncoder.cpp:64) Allocating 1154052 bytes for JPG encoder"
INF/device:plugins:screen:stream 13306 [4100c76794769000] Reading minicap banner
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "INFO: (jni/minicap/minicap.cpp:444) New client connection"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "vc_dnfo = open("/dev/vc-dnfo", O_RDONLY, 0): Invalid argument"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "ioctl_failed = ioctl(vc_dnfo, VC_DNFO_IOC_DISPLAY_INFO, &vc_dnfo_info): Invalid argument"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "vc_dnfo = open("/dev/vc-dnfo", O_RDONLY, 0): Invalid argument"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "ioctl_failed = ioctl(vc_dnfo, VC_DNFO_IOC_DISPLAY_INFO, &vc_dnfo_info): Invalid argument"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "vc_dnfo = open("/dev/vc-dnfo", O_RDONLY, 0): Invalid argument"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "ioctl_failed = ioctl(vc_dnfo, VC_DNFO_IOC_DISPLAY_INFO, &vc_dnfo_info): Invalid argument"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "vc_dnfo = open("/dev/vc-dnfo", O_RDONLY, 0): Invalid argument"
INF/device:plugins:screen:stream 13306 [4100c76794769000] minicap says: "ioctl_failed = ioctl(vc_dnfo, VC_DNFO_IOC_DISPLAY_INFO, &vc_dnfo_info): Invalid argument"

I'm not able to understand the exact issue and it seems that minicap doesnt support this device.

Wanted to check if:
1. you had tested this device ever?
2. can guide me how should I solve this problem.

Regards

**Comment by sorccu** [comment]
1. Nope
2. Every broken device has a different cause & workaround. It's nearly impossible to tell you what you need to do, you have to figure it out yourself. And it's almost impossible to do it remotely. We'd have to spend several hours, perhaps even days with the device. You can become a [hardware sponsor](https://github.com/openstf/stf#sponsors-wanted) if you'd like us to look into it.

Personally I haven't seen that error before, so it would probably take several days to fix.

**Comment by chanchifan** [comment]
@sorccu Hello! I have same issue with Samsung I9152. Did you solve the problem?

**Comment by sorccu** [comment]
No, if someone wants these devices supported, they have to send me one.

Otherwise all you can do is wait.

**Comment by sorccu** [comment]
But please create a new issue for that specific device. That way it's easier to see which devices are not working.

**Comment by sorccu** [comment]
It seems like the I9082 is upgradable to 4.2 (originally 4.1), and @chanchifan's I9152 is 4.2. We've just pushed a fix for many 4.2 devices in STF 2.3.0, could you try and see if those devices work now? And @prateekagarwal60 please let us know which Android version your device was running.

**Comment by sorccu** [comment]
Ah, I can see from the log that it was running on SDK 17, which is 4.2. I believe both of these devices have a very good chance of working now, but please report back with results.
